### PR TITLE
Fix set feature for data blending

### PR DESCRIPTION
**Pull request by x8lucas8x**
This fixes set feature for data blending. I've tested the use cases of using a dimension, a metric and a metric that is also being displayed. Please check tests to see how blending sql is being formatted.

### CODE REVIEW

File: fireant/queries/builder/dataset_blender_query_builder.py [hunk 1]
```diff
@@ -58,56 +61,95 @@ def _find_dataset_fields_needed_to_be_mapped(dataset):
...
 
 
-def _datasets_and_field_maps(blender):
+def _datasets_and_field_maps(blender_dataset, filters):
+    """
+    Returns a list of tuples shaped as dataset and its fields mapped per the blender dataset ones. The field maps
+    are later used for knowing which columns to query on the respective primary and secondary datasets, given the
+    columns selected for the blender dataset.
+
+    :param blender: A DataSetBlender instance.
```
Comment (gl3nn) [inline fireant/queries/builder/dataset_blender_query_builder.py:94]: You probably want to rename this parameter.

Comment (x8lucas8x) [reply fireant/queries/builder/dataset_blender_query_builder.py:94]: Good catch. I'll do.

File: fireant/queries/builder/dataset_blender_query_builder.py [hunk 2]
```diff
@@ -58,56 +61,95 @@ def _find_dataset_fields_needed_to_be_mapped(dataset):
...
+        primary_dataset = dataset.primary_dataset
+        secondary_dataset = dataset.secondary_dataset
+        blender_dataset_fields = apply_set_dimensions(
+            _find_dataset_fields_needed_to_be_mapped(dataset), filters, dataset,
+        )
+
+        primary_dataset_fields = set(apply_set_dimensions(primary_dataset.fields, filters, primary_dataset))
+        secondary_dataset_fields = set(apply_set_dimensions(secondary_dataset.fields, filters, secondary_dataset))
+
+        primary_dimension_per_blender_dimension = dict(_map_field(
```
Comment (gl3nn) [inline fireant/queries/builder/dataset_blender_query_builder.py:111]: Not sure if I prefer this name over `blender_to_primary_field_map`, I find it a bit confusion. Also, does it only contain dimensions?

Comment (x8lucas8x) [reply fireant/queries/builder/dataset_blender_query_builder.py:111]: Not only. Haha, I was renaming it as you commented on this.

**Comment by gl3nn** [comment]
Can you also make sure that you tested it for more than 1 linked dataset? If it works out of the box it probably doesn't need an automated test.

File: fireant/queries/builder/dataset_blender_query_builder.py (hunk 2 above)
Comment (x8lucas8x) [reply fireant/queries/builder/dataset_blender_query_builder.py]: I've reverted that.

File: fireant/queries/builder/dataset_blender_query_builder.py (hunk 1 above)
Comment (x8lucas8x) [reply fireant/queries/builder/dataset_blender_query_builder.py]: Fixed.

**Comment by x8lucas8x** [comment]
> Can you also make sure that you tested it for more than 1 linked dataset? If it works out of the box it probably doesn't need an automated test.

I've added tests with a tertiary dataset being blended.

**Pull request merged**